apps/nextjs/src/components/AppComponents/Chat/chat-lessonPlanDisplay.tsx

```diff
@@ -104,6 +145,27 @@ export const LessonPlanDisplay = ({
   const { userHasCancelledAutoScroll } =
     useDetectScrollOverride(documentContainerRef);
 
+  const sectionRefs = useRef<
+    Partial<
+      Record<LessonPlanKey, React.MutableRefObject<HTMLDivElement | null>>
+    >
+  >({});
+  const setSectionRef = useCallback(
+    (
+      key: LessonPlanKey,
+      ref: React.MutableRefObject<HTMLDivElement | null>,
+    ) => {
+      sectionRefs.current[key] = ref;
+    },
+    [sectionRefs],
+  );
+
+  useSectionScrolling({
+    sectionRefs: sectionRefs.current,
+    documentContainerRef,
+    userHasCancelledAutoScroll,
+  });
```

Comment:
I think this could move to its own logic with a store and a context at some point